rfd/0196-introducing-a-component-system.md

```diff
@@ -0,0 +1,518 @@
+---
+authors: Ryan Clark (ryan.clark@goteleport.com)
+state: draft
+---
+
+# RFD 0194 - Introducing a component system
+
+## Required Approvers
+
+* Engineering: @ravicious || @kimlisa || @avatus
+* Design: @roraback
+
+## What
+
+Introduce Chakra UI, a 3rd-party component system with pre-built components and easy styling capabilities into our web
+UI.
+
+## Why
+
+Modernise our ability to create styled, reusable, useful components without having to reinvent them from scratch each
+time. Speed up development time by having pre-built, tested components ready to use with minimal fuss. Accessibility,
+including ARIA support and keyboard navigation, is built in to all components.
+
+## Current Setup
+
+Our current design system is using `styled-system` and `styled-components` to create both reusable components and all the
+views/pages in the web UI. These are both great libraries, but there are a few drawbacks to the current usage which
+seems them
+being either underused (`styled-system`), or overused (`styled-components`).
+
+### styled-system vs styled-components
+
+Components defined via `styled-system` are essentially "building blocks" and can accept a lot of different props (some
+with shorthand values)
+to define their styles.
+
+```tsx
+<Box
+  px={2}
+  py={1}
+  bg="levels.popout"
+  border="1px solid"
+  borderColor="alpha.200"
+  borderRadius="md"
+>
+  Some content!
+</Box>
+```
+
+Components defined via `styled-components` are more like "display components", typically taking only a few (if not no)
+props and having its style defined
+in a template literal.
+
+```tsx
+const StyledBox = styled.div`
+  padding: ${p => p.theme.space[2]}px ${p => p.theme.space[1]}px;
+  background: ${p => p.theme.colors.levels.popout};
+  border: 1px solid ${p => p.theme.colors.spotBackground[2]};
+  border-radius: ${p => p.theme.radii[2]}px;
+`;
+```
+
+## Drawbacks of current setup
```

Comment:
Another drawback is that styled-system is unmaintained. The [website](https://styled-system.com/) doesn't work anymore and the last commit was 4 years ago.